### PR TITLE
LLSE: New methods and properties in LlAPI

### PR DESCRIPTION
**Pull request by ghost**
## Description

Please carefully read the [Contributing note](https://github.com/LiteLDev/LiteLoaderBDS/discussions/546) before making any pull requests.
And, **Do not make a pull request to merge into main branch unless it is a hotfix. Use the beta branch instead.**
## Changelog

- **LlAPI**
    - Add new methods: `language`, `scriptEngineVersion`, `isDebugMode`, `getVersionStatus`
    - Add new properties `isDev`, `isRelease` in method `version`
    - Add `Plugin` type in documentation (check Basic.md)
    - Add method `getAllPluginInfo`. He return list of plugin object

## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [X] Enhancement
- [X] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/LiteLDev/LiteLoaderBDS/discussions/546)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: ScriptEngine/API/LlAPI.cpp [hunk 1]
```diff
@@ -136,18 +146,51 @@ Local<Value> LlClass::getPluginInfo(const Arguments& args) {
...
+        ver.set("isRelease", LITELOADER_VERSION_STATUS == LL::Version::Status::Release);
+        ver.set("isDev", LITELOADER_VERSION_STATUS == LL::Version::Status::Dev);
         return ver;
     }
     CATCH("Fail in LLSEGetVersion!")
 }
 
+Local<Value> LlClass::getLang(const Arguments& args) {
+    try {
+        return String::newString(LL::globalConfig.language);
```
Comment (Jaciezyt) [inline ScriptEngine/API/LlAPI.cpp]: `LL::globalConfig.language` may be `"system"`, suggest
`PluginOwnData::getImpl<I18nBase>(LL::getLoaderHandle(), I18nBase::POD_KEY).defaultLocaleName`

File: ScriptEngine/API/LlAPI.h [hunk 2]
```diff
@@ -8,13 +8,18 @@ class LlClass {
 public:
     static Local<Value> registerPlugin(const Arguments& args);
     static Local<Value> version(const Arguments& args);
+    static Local<Value> getVersionStatus(const Arguments& args);
+    static Local<Value> getIsDebugMode(const Arguments& args);
+    static Local<Value> getLang(const Arguments& args);
     static Local<Value> versionString(const Arguments& args);
     static Local<Value> requireVersion(const Arguments& args);
-    static Local<Value> listPlugins(const Arguments& args);
+    static Local<Value> listExtPlugins(const Arguments& args);
```
Comment (Jaciezyt) [inline ScriptEngine/API/LlAPI.h]: Why name it `listExtPlugins`? What does `Ext` mean inside?

File: docs/en_US/Development/GameAPI/Basic.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ Currently, LLSE has the following game element objects:
 | `Player`         | Player Object | Identify player or access player attributes       |
 | `Entity`         | Entity Object | Identify entity or access entity attributes      |
 | `Block`          | Block Object  | Identify Block or access block attributes |
+| `Plugin`         | Plugin Object  | Identify any plugin (Native, JS, Lua) |
```
Comment (Jaciezyt) [inline docs/en_US/Development/GameAPI/Basic.md]:
```suggestion
```
This is not a game element obejct

File: docs/en_US/Development/GameAPI/Basic.md [hunk 4]
```diff
@@ -29,6 +30,19 @@ You will come across them frequently in subsequent documentation.
 
 <br>
 
+## 🧩 Plugin Object
+
+| Parameter       | Meaning                              | Data Type  |
```
Comment (Jaciezyt) [inline docs/en_US/Development/GameAPI/Basic.md]:
```suggestion
| Property       | Description                              | Type  |
```

File: docs/en_US/Development/GameAPI/Basic.md [hunk 5]
```diff
@@ -29,6 +30,19 @@ You will come across them frequently in subsequent documentation.
...
+| ------------ | ------------------------------------- | --------- |
+| pl.name    | Plugin name                             | `String` |
+| pl.desc    | Plugin description                      | `String` |
+| pl.ver     | Plugin version (array)                  | `Array<Integer,Integer,Integer>` |
```
Comment (Jaciezyt) [inline docs/en_US/Development/GameAPI/Basic.md]: `pl` may be confused with Player
and `plugin.ver` dosn't match what is written in your code

File: docs/en_US/Development/GameAPI/Basic.md [hunk 6]
```diff
@@ -29,6 +30,19 @@ You will come across them frequently in subsequent documentation.
...
+| pl.versionStr  | Plugin version (string)             | `String` |
+| pl.filePath | Path to plugin                         | `String` |
+| pl.other     | Other information                     | `Object` |
```
Comment (Jaciezyt) [inline docs/en_US/Development/GameAPI/Basic.md]:
```suggestion
| plugin.others     | Other information                     | `Object` |
```

**Review by Jaciezyt — CHANGES_REQUESTED** [review]

File: ScriptEngine/API/LlAPI.h (hunk 2 above)
Comment (ghost) [reply ScriptEngine/API/LlAPI.h]: `Ext` - `extended`. In this method we get extended information about plugins (name, description and etc.)

File: docs/en_US/Development/GameAPI/Basic.md (hunk 5 above)
Comment (ghost) [reply docs/en_US/Development/GameAPI/Basic.md]: > `pl` may be confused with Player and `plugin.ver` dosn't match what is written in your code

**Fixed!**

File: ScriptEngine/API/LlAPI.h (hunk 2 above)
Comment (Jaciezyt) [reply ScriptEngine/API/LlAPI.h]: sounds not good, i prefer to keep `listPlugins`

Comment (ghost) [reply ScriptEngine/API/LlAPI.h]: > sounds not good, i prefer to keep `listPlugins`

okay. and what to do with `listPlugins`, which returns the names of plugins? I suggest calling it `listPluginsName`

Comment (Jaciezyt) [reply ScriptEngine/API/LlAPI.h]: wait, for compatibility, don't change listPlugins, and replace `listExtPlugins` with `getAllPluginInfo`

Comment (ghost) [reply ScriptEngine/API/LlAPI.h]: > wait, for compatibility, don't change listPlugins, and replace `listExtPlugins` with `getAllPluginInfo`

Okay, changed.

File: ScriptEngine/API/LlAPI.cpp (hunk 1 above)
Comment (ghost) [reply ScriptEngine/API/LlAPI.cpp]: > `LL::globalConfig.language` may be `"system"`, suggest `PluginOwnData::getImpl<I18nBase>(LL::getLoaderHandle(), I18nBase::POD_KEY).defaultLocaleName`

**Changed!**

**Review by Jaciezyt — APPROVED** [review]

**Comment by Jaciezyt** [comment]
btw, check [here](https://docs.litebds.com/#/en_US/Maintenance/)

**Pull request merged**